### PR TITLE
CMake build fixes plus a WCSIMDIR related fix

### PR DESCRIPTION
**Pull request by JacekHoleczek**
There are three issues:
* CMake building method improvements (copy all required files to the "external" build directory and make sure that the in-place `Makefile`, if present, has precedence),
* when `WCSIMDIR` is not set, prevent a segmentation fault during the `WCSimDetectorConstruction` and protect the `convertNuPRISM.C` macro,
* small Visual Studio Code related improvements in the `.gitignore` file.

Both building methods tested on CentOS 7 / gcc 4.8.5 and Ubuntu 20.04 / gcc 9.3.0 with ROOT 5 and 6 (and geant4.10.01.p03).

### CODE REVIEW

**Comment by JacekHoleczek** [comment]
I spotted a small macro (`convertNuPRISM.C`) that also did not check if WCSIMDIR was set (now fixed).

**Comment by JacekHoleczek** [comment]
Another small `CMakeLists.txt` fix (removed reference to a nonexistent `${PROJECT_SOURCE_DIR}/../shared/include` directory which may actually exist and contain unrelated stuff).

Everything should be ready now.

**Comment by JacekHoleczek** [comment]
Could you, please, run your automatic tests?

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,32 +91,33 @@ add_executable(WCSim WCSim.cc ${sources} ${headers})
...
-  )
-
-foreach(_script ${WCSIM_SCRIPTS})
-  configure_file(
-    ${PROJECT_SOURCE_DIR}/${_script}
-    ${PROJECT_BINARY_DIR}/${_script}
-    COPYONLY
+if (NOT (PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR))
+  file(COPY ${PROJECT_SOURCE_DIR}/mPMT-configfiles DESTINATION ${PROJECT_BINARY_DIR})
+  file(COPY ${PROJECT_SOURCE_DIR}/macros DESTINATION ${PROJECT_BINARY_DIR})
```
Comment (nickwp) [inline CMakeLists.txt]: Can you explain why this change is necessary? CMake `file(COPY ...)` lines cause problems on our cluster. This makes building many software painful (and the administrators can't / won't make changes to fix it) so I'd rather not make WCSim also painful to build unless it is really necessary.

File: convertNuPRISM.C [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+    if (!(wcsimdirenv && wcsimdirenv[0])) { // make sure it's non-empty
+      wcsimdirenv = "."; // the "default" value
+      std::cout << "Note: WCSIMDIR not set, assuming: " << wcsimdirenv << std::endl;
+      gSystem->Setenv("WCSIMDIR", wcsimdirenv);
+    }
+  }
+  
+  gInterpreter->AddIncludePath("${WCSIMDIR}/include");
+  gSystem->Load("${WCSIMDIR}/libWCSimRoot");
+  gROOT->ProcessLine(".L nuPRISMconvert.cc+");
```
Comment (nickwp) [inline convertNuPRISM.C:13]: Have you checked that these changes to how the library is loaded work with different root versions? Is there a reason for the changes other than preferred style?

**Review by nickwp — COMMENTED** [review]
Sorry, we don't have any automatic tests, so just have to test things manually for now.

File: convertNuPRISM.C (hunk 2 above)
Comment (JacekHoleczek) [reply convertNuPRISM.C:13]: Yes, this part is mainly a source code "polishing" modification. I decided to do it as I anyhow needed to edit this file to protect it against non-existent "`WCSIMDIR`".
Of course, I tested it with both ROOT versions (as stated in my first comment here).

File: CMakeLists.txt (hunk 1 above)
Comment (JacekHoleczek) [reply CMakeLists.txt]: When the CMake building method is used in a separate directory (i.e., not when building "in-place"), one needs to copy two directories (which are then required at run-time): "`mPMT-configfiles`" and "`macros`"
In the old "`CMakeLists.txt`" file, there was only a partial explicit list of files from "`macros`" to be copied.
Of course, one could again make a full explicit list of all currently existing files and put it into the "`CMakeLists.txt`" file.
However, when people add / remove files to / from these directories, nobody really remembers to update this fixed list in this file.
As a result, one either has missing files (when some files have been added) or the CMake dies (when some files have been deleted or simply renamed).

So, it seems to me that the best idea is to make sure that always entire directories are copied automatically (using `file(COPY ...)` lines).

Could you, please, tell me what problems you face with these `file(COPY ...)` lines?

If they do not work for you, could you, please, try an equivalent form which is also expected to work (tested by me, of course, too):
```
  file(COPY ./mPMT-configfiles DESTINATION ./ )
  file(COPY ./macros DESTINATION ./ )
```

Comment (nickwp) [reply CMakeLists.txt]: The problem we have with these commands is due to how the filesystem quota works on our cluster. When creating files in the shared project space they must belong to the correct group, not just the user creating the file. Usually the filesystem handles this and we don't need to worry, but some CMake commands create files without the correct group ownership for some reason.
Anyway, after some investigating, this command can be made to work in our situation by also specifying `USE_SOURCE_PERMISSIONS`, i.e.:
```
  file(COPY ${PROJECT_SOURCE_DIR}/mPMT-configfiles DESTINATION ${PROJECT_BINARY_DIR} USE_SOURCE_PERMISSIONS)
  file(COPY ${PROJECT_SOURCE_DIR}/macros DESTINATION ${PROJECT_BINARY_DIR} USE_SOURCE_PERMISSIONS)
```
This should actually be the default for copying files, but perhaps it depends on cmake version or something.
Could you update the commands to include this, then I think it all should be fine.

**Comment by JacekHoleczek** [comment]
I added `USE_SOURCE_PERMISSIONS` to all `file(COPY ...)` calls.

File: CMakeLists.txt (hunk 1 above)
Comment (JacekHoleczek) [reply CMakeLists.txt]: pull request updated (see below)

**Review by nickwp — APPROVED** [review]
